src/main/java/com/codeus/winter/config/DefaultBeanFactory.java

```diff
@@ -245,4 +245,60 @@ private Object resolveConstructor(Class<?> beanClass) {
         }
         throw new BeanFactoryException("Unable to resolve dependencies for class: " + beanClass.getName());
     }
+
+    private Object getBeanDependency(Class<?> dependencyClass, Class<?> beanClass) {
```

Comment:
I'm not sure if the implementation covers one edge case, could you check:

```
class A {
}

class B {
  // Inject here
  Collection<A> collection
}

```
and pass the bean definitions in the order [BeanDefinitionA_First, BeanDefinitionB, BeanDefinitionA_Second].
Will both instances be injected into BeanDefinitionB?
I believe it works, but please add one test case for this situation